src/Keboola/StorageApi/Downloader/S3Downloader.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Keboola\StorageApi\Downloader;
+
+use Aws\S3\S3Client;
+
+class S3Downloader implements DownloaderInterface
+{
+    /**
+     * @var S3Client
+     */
+    private $client;
+
+    /**
+     * S3Downloader constructor.
+     */
+    public function __construct(S3Client $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function downloadFileFromFileResponse($fileResponse, $tmpFilePath)
+    {
+        $this->client->getObject([
+            'Bucket' => $fileResponse["s3Path"]["bucket"],
+            'Key' => $fileResponse["s3Path"]["key"],
+            'SaveAs' => $tmpFilePath,
+        ]);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function downloadManifestEntry($fileResponse, $entry, $tmpFilePath)
+    {
+        $fileKey = substr($entry["url"], strpos($entry["url"], '/', 5) + 1);
```

Comment:
Nedokážu rozluštit, co to dělá. Takže bych to vytahnul do statické metody tady a tu otestoval.

Reply:
(myšleno i ten řádek pod tím...)

Reply:
Pocitově to vezme všechno za pátým lomítkem. Ale proč za pátým? 
Proč se replacuje / za _ když se z toho pak dělá md5?

Reply:
A proč vůbec md5? Proč ne uniquid?

Reply:
Je to původní kód, taky netuším co to dělá a proč. Já potřebuju podporu ABS a aby to fungovalo stejně, jakože do statické metody to može byt, ale kdyby tam bylo uniquid tak to neotestuješ leda přes nějaký reqex což je zbytečné.

Reply:
Jo už čtu další koment :D